packages/by-name/qemu-tdx-static/0004-hw-x86-load-initrd-to-static-address.patch

```diff
@@ -43,42 +25,10 @@ diff --git a/hw/i386/x86.c b/hw/i386/x86.c
 index 504575abfa98bc25e498e219a2d58d8d31e5feaa..0763462c16f4106d0aa6a46c2b9c360e36ae3e96 100644
 --- a/hw/i386/x86.c
 +++ b/hw/i386/x86.c
-@@ -953,6 +953,41 @@ void x86_load_linux(X86MachineState *x86ms,
+@@ -953,6 +953,9 @@ void x86_load_linux(X86MachineState *x86ms,
          initrd_max = x86ms->below_4g_mem_size - acpi_data_size - 1;
      }
  
-+    /*
-+     * For TDX RTMRs to be predictable regardless of VM memory size, we need to
-+     * load the initrd to a static address, so no dynamic value ends up in the
-+     * mapped kernel image.
-+     *
-+     * Without setting this to a static address, the address the initrd is mapped to
-+     * which depend not only on the size of the initrd, but also on the memory space
-+     * of the guest, this is not viable for Contrast's reference-value-based attestation
-+     * approach.
-+     *
-+     * As we control the minimum VM memory size in Contrast, we just load the initrd
-+     * to the address it gets loaded to for Contrast's minimum VM memory (2Gi), regardless
-+     * of if the VM has more memory.
-+     *
-+     * QEMU, by default, does a similar thing.
-+     * Consider the below line (cited from above):
-+     *
-+     *    initrd_max = x86ms->below_4g_mem_size - acpi_data_size - 1;
-+     *
-+     * This adds an artifical upper bound of where the initrd can be loaded to, as
-+     * the calculation is based on the VM memory (below_4g_mem_size), but capped at 4Gi.
-+     * This means, the initrd, regardless of guest memory size, will always be loaded
-+     * at address 0x100000000 (4Gi) max (minus ACPI data size).
-+     *
-+     * Essentially, overwriting this to 0x80000000 (2Gi), we create an artificial lower *and*
-+     * upper bound (set to Contrast minimum TDX VM memory size).
-+     * This means that the initrd will *always* be loaded at 0x80000000 (2Gi), minus ACPI
-+     * data size. The difference to QEMU's setting is, that we *fix* the address, rather than
-+     * setting *only* an upper bound.
-+     *
-+     * This way, we get the initrd to *always* be loaded at a static address.
-+     */
 +    uint32_t contrast_min_memory = 0x80000000; // 2Gi
 +    initrd_max = contrast_min_memory - acpi_data_size - 1;
 +
```

Comment:
Should we make this more coherent with the (new) comment by actually setting it to a static address?

Reply:
Are you referring to the `acpi_data_size`? That' a constant afaict: https://github.com/qemu/qemu/blob/495de0fd82d8bb2d7035f82d9869cfeb48de2f9e/hw/i386/pc.c#L288 (or 0 maybe).

Reply:
No, I mean setting `initrd_max` (or `initrd_addr` directly) to something like `0xFFFFF` or some other constant. I'm fine with keeping it as-is too, but having something being called `contrast_min_memory` even though there is no minimum requirement "anymore" bugs me a little here.

Reply:
Done